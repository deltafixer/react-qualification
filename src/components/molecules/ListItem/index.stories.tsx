import React from "react";
import ListItem, { ListItemProps } from "./";

export default {
  component: ListItem,
  title: "ListItem",
  // excludes dummy objects
  excludeStories: /.*Data$/,
};

const propsData = {
  name: "Random name",
  avatar_url: "https://avatars3.githubusercontent.com/u/69631?v=4",
  created_at: new Date().toString(),
  updated_at: new Date().toString(),
};

// hardcoding styled div wrapper because storybooks theme is dark, app is white
// would make it smarter if it wasn't a test project (not to go deep in this when no need)

export const Default = () => (
  <div style={{ background: "white", width: "100%", margin: "auto" }}>
    <ListItem {...(propsData as ListItemProps)} />
  </div>
);
