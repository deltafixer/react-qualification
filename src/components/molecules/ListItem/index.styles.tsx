import styled from "styled-components";

export const StyledListItem = styled.li`
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 12px;

  margin-bottom: 3px;

  > *:not(:last-child) {
    margin-right: 6px;
  }

  > img {
    width: 24px;
  }

  .dates {
    display: flex;
    flex-direction: column;
    font-size: 12px;
    margin: 0 0 0 auto;
    align-items: flex-end;
  }
`;
