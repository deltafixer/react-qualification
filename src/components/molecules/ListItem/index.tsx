import React from "react";
import { StyledListItem } from "./index.styles";

export type ListItemProps = {
  name: string;
  avatar_url: string;
  created_at: string;
  updated_at: string;
};

const ListItem = ({
  name,
  avatar_url,
  created_at,
  updated_at,
}: ListItemProps): JSX.Element => {
  return (
    <StyledListItem>
      <img src={avatar_url} alt="" />
      <span>{name}</span>
      <div className="dates">
        <span>Created at: {new Date(created_at).toLocaleString()}</span>
        <span>Updated at: {new Date(updated_at).toLocaleString()}</span>
      </div>
    </StyledListItem>
  );
};

export default ListItem;
