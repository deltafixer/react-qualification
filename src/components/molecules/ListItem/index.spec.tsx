import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import ListItem, { ListItemProps } from "./index";

let wrapper: ShallowWrapper;

Date.now = jest.fn(() => 1482363367071);

const props: ListItemProps = {
  name: "Random name",
  avatar_url: "https://avatars3.githubusercontent.com/u/69631?v=4",
  created_at: Date.now().toString(),
  updated_at: Date.now().toString(),
};

describe("ListItem component tests", () => {
  beforeAll(() => {
    wrapper = shallow(<ListItem {...props} />);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
