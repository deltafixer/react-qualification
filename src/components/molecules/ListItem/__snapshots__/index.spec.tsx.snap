// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component tests renders correctly 1`] = `
"<styled.li>
  <img src=\\"https://avatars3.githubusercontent.com/u/69631?v=4\\" alt=\\"\\" />
  <span>
    Random name
  </span>
  <div className=\\"dates\\">
    <span>
      Created at: 
      Invalid Date
    </span>
    <span>
      Updated at: 
      Invalid Date
    </span>
  </div>
</styled.li>"
`;
